src/test/java/com/contaazul/dsl/DateDslTest.java

```diff
@@ -116,6 +116,56 @@ public void firstDayOfNextMonth() {
 	}
 
 	@Test
+	public void firstDayOfQuarter() {
+		assertFirstDayOfQuarter( Calendar.JANUARY, Calendar.JANUARY );
+		assertFirstDayOfQuarter( Calendar.FEBRUARY, Calendar.JANUARY );
+		assertFirstDayOfQuarter( Calendar.MARCH, Calendar.JANUARY );
+		assertFirstDayOfQuarter( Calendar.APRIL, Calendar.APRIL );
+		assertFirstDayOfQuarter( Calendar.MAY, Calendar.APRIL );
+		assertFirstDayOfQuarter( Calendar.JUNE, Calendar.APRIL );
+		assertFirstDayOfQuarter( Calendar.JULY, Calendar.JULY );
+		assertFirstDayOfQuarter( Calendar.AUGUST, Calendar.JULY );
+		assertFirstDayOfQuarter( Calendar.SEPTEMBER, Calendar.JULY );
+		assertFirstDayOfQuarter( Calendar.OCTOBER, Calendar.OCTOBER );
+		assertFirstDayOfQuarter( Calendar.NOVEMBER, Calendar.OCTOBER );
+		assertFirstDayOfQuarter( Calendar.DECEMBER, Calendar.OCTOBER );
+	}
+
+	private void assertFirstDayOfQuarter(int currentMonth, int expectedMonth) {
+		Calendar cal = Calendar.getInstance();
+		cal.set( Calendar.MONTH, currentMonth );
+		DateBuilder firstDayOfQuarter = date( cal ).firstDayOfQuarter();
+
+		assertEquals( expectedMonth, firstDayOfQuarter.getMonth() );
+		assertEquals( 1, firstDayOfQuarter.getDayOfMonth() );
+	}
+
+	@Test
+	public void lastDayOfQuarter() {
+		assertLastDayOfQuarter( Calendar.JANUARY, Calendar.MARCH );
+		assertLastDayOfQuarter( Calendar.FEBRUARY, Calendar.MARCH );
+		assertLastDayOfQuarter( Calendar.MARCH, Calendar.MARCH );
+		assertLastDayOfQuarter( Calendar.APRIL, Calendar.JUNE );
+		assertLastDayOfQuarter( Calendar.MAY, Calendar.JUNE );
+		assertLastDayOfQuarter( Calendar.JUNE, Calendar.JUNE );

```

Comment:
Is the last day of the second quarter 31? June ends on 30. Also September ends on 30.